test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -51,11 +57,86 @@
+          jruby-service (reify jruby/JRubyPuppetService
+                          (borrow-instance [_ _] jruby-instance)
+                          (return-instance [_ jruby _]
+                            (reset! returned-jruby jruby))
+                          (get-environment-class-info [_ jruby env]
+                            (reset! get-env-jruby jruby)
+                            (if (= env "production")
+                              {"/one/file"
+                               [
+                                {"name" "oneclass",
+                                 "params" [
+                                           {"name" "oneparam",
+                                            "type" "String",
+                                            "default_literal" "'literal'",
+                                            "default_source" "literal"},
+                                           {"name" "twoparam",
+                                            "type" "Integer",
+                                            "default_literal" "3",
+                                            "default_source" "3"}]
+                                 },
+                                {"name" "twoclass"
+                                 "params" []}],
+                               "/two/file" []})))
+          handler (fn ([req] {:request req}))
+          app (build-ring-handler handler "1.2.3" jruby-service)
+          request (partial app-request app)]
+      (testing "returns good data for environment that exists"
+        (let [response (request
+                        "/v3/environment_classes?environment=production")
+              body-from-json (cheshire/parse-string (:body response))]
+          (is (= 200 (:status response))
+              "Unexpected status code returned from request")
+          (is (= {"name" "production",
+                  "files" [
+                           {"path" "/one/file",
+                            "classes" [{
+                                        "name" "oneclass"
+                                        "params" [
+                                                  {"name" "oneparam",
+                                                   "type" "String",
+                                                   "default_literal" "'literal'",
+                                                   "default_source" "literal"},
+                                                  {"name" "twoparam",
+                                                   "type" "Integer",
+                                                   "default_literal" "3",
+                                                   "default_source" "3"}]},
+                                       {
+                                        "name" "twoclass"
+                                        "params" []}]},
+                           {"path" "/two/file"
+                            "classes" []}]}
+                 body-from-json)
```

Comment:
What value do you think these tests add beyond the integration tests?  I know that I've been advocating for having more unit tests, but there's so much setup logic here and so much mocking that I'm not sure it is really worth it.